src/Compilers/CSharp/Portable/CSharpCompilationOptions.cs

```diff
@@ -42,6 +42,17 @@ public sealed class CSharpCompilationOptions : CompilationOptions, IEquatable<CS
         /// </summary>
         public override NullableContextOptions NullableContextOptions { get; protected set; }
 
+        /// <summary>
+        /// The warning version, which indicates which set of warnings should be produced by the compiler.
+        /// </summary>
+        /// <remarks>
+        /// Most warnings are produced by the compiler unconditionally, but warnings that are controlled by a version
+        /// are only produced when the WarningVersion value is greater than or equal to the warning version
+        /// associated with the warning. Use zero to get none of the versioned warnings, or a large number like 9999
+        /// to get all of the versioned warnings. The first version that enables some warnings is version 5.
+        /// </remarks>
+        public decimal WarningVersion { get; private set; }
```

Comment:
Why `decimal` and not `Version`, given the property itself is named version?

Reply:
[https://github.com/dotnet/roslyn/issues/45941](https://github.com/dotnet/roslyn/issues/45941)

---
In reply to: [453250225](https://github.com/dotnet/roslyn/pull/45750#discussion_r453250225) [](ancestors = 453250225)